### PR TITLE
Unicode

### PR DESCRIPTION
**Pull request by casperdcl**
Fallback ascii if not specified and not explicitly supported by stream

### CODE REVIEW

**Comment by casperdcl** [comment]
fixes #15

File: tqdm/_tqdm.py [hunk 1]
```diff
@@ -26,13 +26,14 @@ def format_interval(t):
...
-def format_meter(n, total, elapsed, ncols=None, prefix=''):
+def format_meter(n, total, elapsed, ncols=None, prefix='', ascii=False):
     # n - number of finished iterations
     # total - total number of iterations, or None
     # elapsed - number of seconds passed since start
     # ncols - the output width in chars. If specified, dynamically resizes bar.
     #     [default bar width: 10].
     # prefix - prepend message (included in total width)
+    # ascii - whether to use ascii (otherwise unicode) character set
     if total and n > total:
```
Comment (hadim) [inline tqdm/_tqdm.py]: Could you add a proper doctstring for parameters please ?

**Comment by hadim** [comment]
Thanks for working on this !

**Comment by casperdcl** [comment]
lol tbh I'm just using this as procrastination. It might look like I've been working on this all day for the last few days but really I haven't. I work better when constantly switching projects :P

**Comment by casperdcl** [comment]
sheesh. @kmike late to the party, yes, maybe, but definitely partyin'

```
Total number of files: 14
Total number of lines: 586
Total number of commits: 60
+----------------------+-----+---------+-------+--------------------+
| name                 | loc | commits | files | distribution       |
+----------------------+-----+---------+-------+--------------------+
| Casper da Costa-Luis | 219 | 15      | 4     | 37.4 / 25.0 / 28.6 |
| Hadrien Mary         | 97  | 8       | 6     | 16.6 / 13.3 / 42.9 |
| Mikhail Korobov      | 75  | 10      | 9     | 12.8 / 16.7 / 64.3 |
| Ivan Ivanov          | 68  | 11      | 4     | 11.6 / 18.3 / 28.6 |
| noamraph             | 65  | 3       | 3     | 11.1 / 5.0 / 21.4  |
| Noam Yorav-Raphael   | 55  | 8       | 2     | 9.4 / 13.3 / 14.3  |
| arkottke             | 5   | 1       | 1     | 0.9 / 1.7 / 7.1    |
| stonebig             | 2   | 1       | 1     | 0.3 / 1.7 / 7.1    |
| Ford Hurley          | 0   | 1       | 0     | 0.0 / 1.7 / 0.0    |
| Jack McCracken       | 0   | 2       | 0     | 0.0 / 3.3 / 0.0    |
+----------------------+-----+---------+-------+--------------------+
```

File: tqdm/_tqdm.py [hunk 2]
```diff
@@ -133,10 +146,23 @@ def tqdm(iterable, desc=None, total=None, leave=False, file=sys.stderr,
...
+            file.encoding
+        except AttributeError:
+            ascii = True
+        else:
+            if file.encoding:
+                ascii = not ('U8' == file.encoding or
+                             ('utf' in file.encoding) or
+                             ('UTF' in file.encoding))
+            else:
+                ascii = True
```
Comment (kmike) [inline tqdm/_tqdm.py]: What about extracting this to a helper function? Something like `file_suports_unicode`:

```
def _is_utf(encoding):
    return ('U8' == encoding) or ('utf' in encoding) or ('UTF' in encoding))

def file_supports_unicode(fp):
    if not getattr(fp, 'encoding', None):
        return False    
    return _is_utf(file.encoding)
```

Comment (hadim) [inline tqdm/_tqdm.py]: +1

Comment (casperdcl) [inline tqdm/_tqdm.py]: will leave it to you ;) i'm not pedantic about loop startup overhead so don't mind abstractions at this level.

**Comment by casperdcl** [comment]
haha @kmike rebase, sure. I've done it before -- #11 . But you still have to do your `file_suports_unicode`

**Comment by casperdcl** [comment]
nvm. was easier to correct than type chat message about it lol

**Comment by casperdcl** [comment]
going to sleep, then spend weekend at a data science competition hackathon (used `nltk` a LOT in the buildup stages, you might be interested to know, @kmike )

**Pull request merged**